lib/connector/connect.js

```diff
@@ -8,8 +8,8 @@ const _logger = require('logging-facility').getLogger('connector');
 async function connect(devicename, baudrate, applyConnectionCheck=true, connectDelay=0){
     // handle connection errors (device not available, permission errors..)
     try{
-        // new terminal line sequence "\n\r>"
```

Comment:
is this still backward compatible to esp8266 ?

Reply:
Yes, the utility works both with esp8266 and esp32. But I only tested esp8266 firmware version 3.0.0.
I didn't test version 1.5.4 because it's very old.

```bash

❯ ./nodemcu-tool mkfs
[NodeMCU-Tool]~ Do you really want to format the filesystem and delete all file ? (no) yes
[NodeMCU-Tool]~ Connected 
[device]      ~ Model: esp8266 | Version: 3.0.0 | ChipID: 0x827f58 | FlashID: 0x164020 
[device]      ~ Formatting the file system...this will take around ~30s 
[device]      ~ File System created | ....................................................format done. 
[NodeMCU-Tool]~ disconnecting 
❯ ./nodemcu-tool upload helloworld.lua
[NodeMCU-Tool]~ Connected 
[device]      ~ Model: esp8266 | Version: 3.0.0 | ChipID: 0x827f58 | FlashID: 0x164020 
[NodeMCU-Tool]~ Uploading "helloworld.lua" >> "helloworld.lua"... 
[connector]   ~ Transfer-Mode: base64 
[NodeMCU-Tool]~ File Transfer complete! 
[NodeMCU-Tool]~ disconnecting 
❯ ./nodemcu-tool fsinfo
[NodeMCU-Tool]~ Connected 
[device]      ~ Model: esp8266 | Version: 3.0.0 | ChipID: 0x827f58 | FlashID: 0x164020 
[device]      ~ Free Disk Space: 3107 KB | Total: 3108 KB | 1 Files 
[device]      ~ Files stored into Flash (SPIFFS) 
[device]      ~  - helloworld.lua (383 Bytes) 
[NodeMCU-Tool]~ disconnecting 

```

Firmware for esp32 was tested on esp32c3, esp32s3 zero, nano esp32c6 v1.0 and esp32 devkit v4

```bash
❯ ./nodemcu-tool -p /dev/ttyACM0 fsinfo
[NodeMCU-Tool]~ Connected 
[device]      ~ Model: esp32c3 | Version: 0.0.0 | ChipID: 0x_unknown | FlashID: 0x204016 
[device]      ~ Free Disk Space: 390 KB | Total: 404 KB | 2 Files 
[device]      ~ Files stored into Flash (SPIFFS) 
[device]      ~  - loremipsum.lua (13313 Bytes) 
[device]      ~  - helloworld.lua (383 Bytes) 
[NodeMCU-Tool]~ disconnecting 

❯ ./nodemcu-tool -p /dev/ttyACM0 fsinfo
[NodeMCU-Tool]~ Connected 
[device]      ~ Model: esp32s3 | Version: 0.0.0 | ChipID: 0x_unknown | FlashID: 0x204016 
[device]      ~ Free Disk Space: 390 KB | Total: 404 KB | 2 Files 
[device]      ~ Files stored into Flash (SPIFFS) 
[device]      ~  - loremipsum.lua (13313 Bytes) 
[device]      ~  - helloworld.lua (383 Bytes) 
[NodeMCU-Tool]~ disconnecting 

❯ ./nodemcu-tool -p /dev/ttyUSB0 fsinfo
[NodeMCU-Tool]~ Connected 
[device]      ~ Model: esp32 | Version: 0.0.0 | ChipID: 0x2ac82e182298 | FlashID: 0x684016 
[device]      ~ Free Disk Space: 403 KB | Total: 404 KB | 2 Files 
[device]      ~ Files stored into Flash (SPIFFS) 
[device]      ~  - helloworldLF.lua (389 Bytes) 
[device]      ~  - node_info.lua (377 Bytes) 
[NodeMCU-Tool]~ disconnecting 

```